### PR TITLE
ECS C interop support.

### PR DESCRIPTION
**Pull request by RubyNova**
This PR introduces the C interop support for NovelRT::Ecs with relevant engine core changes.

### CODE REVIEW

File: include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    NrtComponentBufferMemoryContainer_ImmutableDataView Nrt_ComponentBufferMemoryContainer_GetComponentUnsafe(
+        NrtComponentBufferMemoryContainer container,
+        NrtEntityId entity);
+
+    NrtBool Nrt_ComponentBufferMemoryContainer_HasComponent(NrtComponentBufferMemoryContainer container,
+                                                            NrtEntityId entity);
+
+    size_t Nrt_ComponentBufferMemoryContainer_GetImmutableDataLength(NrtComponentBufferMemoryContainer container);
+
+    // clang-format off
```
Comment (capnkenny) [inline include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h:53]: Was there a reason for turning formatting off here?

File: include/NovelRT.Interop/Ecs/NrtEcs.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
```
Comment (capnkenny) [inline include/NovelRT.Interop/Ecs/NrtEcs.h:3]: (I cant drag for multilines on mobile x.x)
Is this file necessary?

File: include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h (hunk 1 above)
Comment (capnkenny) [inline include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h:53]: Is there a reason for turning formatting off here?

File: include/NovelRT.Interop/Ecs/NrtSparseSetMemoryContainer.h [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    typedef struct SparseSetByteIteratorViewHandle* NrtSparseSetMemoryContainer_ByteIteratorView;
+
+    typedef struct SparseSetConstByteIteratorViewHandle* NrtSparseSetMemoryContainer_ConstByteIteratorView;
+
+    typedef struct SparseSetIteratorHandle* NrtSparseSetMemoryContainer_Iterator;
+
+    typedef struct SparseSetConstIteratorHandle* NrtSparseSetMemoryContainer_ConstIterator;
+
+NrtSparseSetMemoryContainer Nrt_SparseSetMemoryContainer_Create(size_t sizeOfDataTypeInBytes);
```
Comment (capnkenny) [inline include/NovelRT.Interop/Ecs/NrtSparseSetMemoryContainer.h:24]: Dunno how this got past formatting :)

**Review by capnkenny — COMMENTED** [review]
Looked through Interop and it seems sane - going to give the rest a review in a while since there's a lot here. Only a few questions for now!

File: include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h (hunk 1 above)
Comment (RubyNova) [reply include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h:53]: I've been doing it in any place where we have begin() and end() functions, but since its a clang-tidy thing now, not really sure if this does what I intended.

File: include/NovelRT.Interop/Ecs/NrtEcs.h (hunk 2 above)
Comment (RubyNova) [reply include/NovelRT.Interop/Ecs/NrtEcs.h:3]: Yes, its the namespace file, I've just not filled it in yet with the relevant includes.

File: include/NovelRT.Interop/Ecs/NrtSparseSetMemoryContainer.h (hunk 3 above)
Comment (RubyNova) [reply include/NovelRT.Interop/Ecs/NrtSparseSetMemoryContainer.h:24]: Just CLion things, but also, I've not been trying that hard since CLion disagrees with CI anyway lmao

File: include/NovelRT.Interop/Ecs/NrtEcs.h (hunk 2 above)
Comment (capnkenny) [reply include/NovelRT.Interop/Ecs/NrtEcs.h:3]: Okayyyy just making sure it didn't slip or anything! 😄

File: include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h (hunk 1 above)
Comment (capnkenny) [reply include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h:53]: Agreed, you may be same to remove this as the definition itself shouldn't be affected by clang-format.

If It breaks afterwards we can always put it back

File: include/NovelRT.Interop/Ecs/NrtSparseSetMemoryContainer.h (hunk 3 above)
Comment (capnkenny) [reply include/NovelRT.Interop/Ecs/NrtSparseSetMemoryContainer.h:24]: Ohhh CLion....

**Review by capnkenny — APPROVED** [review]
LGTM - you may want to add a ticket as a reminder for extra documentation (unless I misunderstood you on Discord :^) )
